core/src/core-api/auth.js

```diff
@@ -0,0 +1,31 @@
+// import * as AsyncHelpers from '../utilities/helpers/async-helpers';
+// import * as GenericHelpers from '../utilities/helpers/generic-helpers';
+import { config } from './config';
+
+class LuigiAuthManager {
+  constructor() { }
+
+  /*
+   * Detects if authorization is enabled via configuration.
+   * @returns {boolean} returns true if authorization is enabled. Otherwise returns false.
+   */
+  isAuthorizationEnabled() {
+    const idpProviderName = config.getConfigValue('auth.use');
+    const idpProviderSettings = config.getConfigValue(
+      `auth.${idpProviderName}`
+    );
+    return !!idpProviderSettings;
+  }
+
+  async handleAuthEvent(eventName, providerInstanceSettings, data, redirectUrl) {
+    const result = await config.executeConfigFnAsync('auth.events.' + eventName, providerInstanceSettings, data);
```

Comment:
If the event handler function is not defined this will lead to an error in browser console: 
![Bildschirmfoto 2019-05-06 um 11 18 16](https://user-images.githubusercontent.com/11227688/57216677-d3566700-6ff0-11e9-975a-87f4cf4c8522.png)
As these event handlers are optional this should not lead to an error. We either test first if there is an event handler fn defined and then execute it here or we build this check into the executeConfigFnAsync; maybe with an 'optional' flag so that it fails if the function is required in cases where it is required.